backend/analytics_server/tests/service/Incidents/test_team_pr_incidents.py

```diff
@@ -0,0 +1,236 @@
+        return self._prs_using_filters
+
+    def get_prs_merged_in_interval_by_numbers(self, *args, **kwargs):
+        return self._prs_using_numbers
+
+
+class FakeIncidentsRepoService:
+    pass
+
+
+def test_get_team_pr_incidents_no_filters():
+    incident_service = get_incident_service()
+
+    mock_settings_service = Mock()
+    mock_settings_service.get_settings.return_value.specific_settings = (
+        IncidentPRsSetting(include_revert_prs=True, filters=[])
+    )
+    incident_service._settings_service = mock_settings_service
+
+    result = incident_service.get_team_pr_incidents(
+        "team_1",
+        mock_interval(),
+        PRFilter(),
+    )
+
+    assert result == []
+
+
+def test_get_team_pr_incidents_with_filters():
+
+    prs_using_filters = [
+        PullRequest(
+            id="pr_2_of_repo_1",
+            repo_id="repo_1",
+            number="2",
+            head_branch="revert-1",
+        ),
+        PullRequest(
+            id="pr_4_of_repo_1",
+            repo_id="repo_1",
+            number="4",
+            head_branch="branch_4",
+            title="Revert PR #3 due to some reason",
+        ),
+    ]
+
+    prs_using_numbers = [
+        PullRequest(
+            id="pr_1_of_repo_1", repo_id="repo_1", number="1", head_branch="branch_1"
+        ),
+        PullRequest(
+            id="pr_3_of_repo_1",
+            repo_id="repo_1",
+            number="3",
+            head_branch="branch_3",
+        ),
+    ]
+
+    incident_service = IncidentService(
+        FakeIncidentsRepoService(),
+        FakeSettingsService(),
+        FakeCodeRepoService(prs_using_filters, prs_using_numbers),
+    )
+
+    expected_result_keys = [
+        "pr_1_of_repo_1",
+        "pr_3_of_repo_1",
+    ]
+
+    result = incident_service.get_team_pr_incidents(
+        "team_1",
+        mock_interval(),
+        PRFilter(),
+    )
+
+    assert expected_result_keys == [incident.key for incident in result]
+
+
+def test_get_team_pr_incidents_with_multiple_repos_but_no_incidents():
+
+    prs_using_filters = [
+        PullRequest(
+            id="pr_2_of_repo_1",
+            repo_id="repo_1",
+            number="2",
+            head_branch="revert-1",
+        ),
+        PullRequest(
+            id="pr_2_of_repo_2",
+            repo_id="repo_2",
+            number="2",
+            head_branch="revert-1",
+        ),
+    ]
+
+    prs_using_numbers = []
+
+    incident_service = IncidentService(
+        FakeIncidentsRepoService(),
+        FakeSettingsService(),
+        FakeCodeRepoService(prs_using_filters, prs_using_numbers),
+    )
+
+    expected_result_keys = []
+
+    result = incident_service.get_team_pr_incidents(
+        "team_1",
+        mock_interval(),
+        PRFilter(),
+    )
+
+    assert expected_result_keys == [incident.key for incident in result]
+
+
+def test_get_team_pr_incidents_with_multiple_repos_and_incidents():
+
+    prs_using_filters = [
```

Comment:
```suggestion
    resolution_prs = [
```
can be named better to signify what exactly those prs are
similar for other variables in the test

Reply:
My bad. This should have been named `resolution_prs` as they are fetched using our PR Incident filters.

Reply:
Also I shouldn't rename `prs_using_numbers` right ?
Because this PRs are Incident PRs is not confirmed yet.

Ex: Resolution PR revert `123` and we have 3 Repos.
Then we try to fetch PR number `123` from each Repo.
So len(prs_using_numbers) = 3 but only one of these will be Incident PR.

Reply:
sure